src/services/SendATFReport.ts

```diff
@@ -53,15 +55,18 @@ class SendATFReport {
   /**
    * Method to collate testResults and waitActivities into a common list
    * and then sort them on startTime to display the activities in a sequence.
-   * @param testResultsList: testResults list
-   * @param waitActivitiesList: wait activities list
+   * @param testResultsList
+   * @param waitActivitiesList
    */
-  public computeActivitiesList(testResultsList: ITestResults[], waitActivitiesList: IActivity[]) {
+  public computeActivitiesList(testResultsList: TestResultSchema[], waitActivitiesList: ActivitySchema[]) {
     const list: IActivitiesList[] = [];
+
     // Adding Test activities to the list
     for (const testResult of testResultsList) {
+      const testResultTestType = testResult.testTypes as unknown as TestTypeSchema;
```

Comment:
Now that we are using the corrected version of TestResultSchema, wouldn't this be TestTypeSchema[ ]?

```
export interface TestResultSchema {
  ...
  testTypes: TestTypeSchema[];
  ...
}
```

In ITestResults previously, it was incorrect defined as a singular object.

```
interface ITestResults {
  ...
  testTypes: ITestType;
  ...
}
```

Reply:
It was previously an object before this, why would it now be an array?

Reply:
The Test Result object has always an array of testTypes. It looks like in this service, and others the definition was incorrect.

Reply:
How is it working now, if it's an array but is expecting it to be an object?

Reply:
I'm really not sure, maybe it works when there is only one test type in the array, and causes issues if there is more than one? The test result examples used for the tests in this repo only have 1 testType in them, but clearly, as you can see, they are in an array of testTypes.

Reply:
The expand test results is converting the single test result with multiple test types into multiple test results with a single test type in each. Which is why it's coming out as a single object and without changing that process I think that we should keep this as the single object.

Reply:
I see your point and the need to split each test type out. But the reason this ticket was scored the way it was, is because there was a possibility implemementing stricter types would result in changes having to be made (the exact same thing will need to be done in cert-gen as that also splits out the test-types into singular objects).

Leaving it as you currently have it would mean the expandTestResults method having to return any[ ] which is something we are trying to get away from with all of the changes to types definition and the introduction of this TestResultSchema.

```
testResult.testTypes.forEach((testType: TestTypeSchema, i: number, array: any[]) => {
    const clonedRecord: TestResultSchema = Object.assign({}, templateRecord); // Create test result from template
    Object.assign(clonedRecord, { testTypes: testType }); // Assign it the test type

    splittedRecords.push(clonedRecord);
});
```

The solution may be as simple assigning the testType to an array in the splittedRecords and just grabbing the first one in the subsequent logic, that way we know exactly what objects we are dealing with.

```
Object.assign(clonedRecord, { testTypes: [testType] }); // Assign it the test type in an array
```

Let's all have a chat after stand-up so we can discuss further and go through any concerns.